pkg/controllers/dynakube/connectioninfo/reconciler.go

```diff
@@ -47,16 +47,25 @@ func (r Reconciler) updateDynakubeStatus(ctx context.Context) error {
 	return nil
 }
 
-func (r *Reconciler) Reconcile(ctx context.Context) error {
+func (r *Reconciler) ReconcileAG(ctx context.Context) error {
 	oldStatus := r.dynakube.Status.DeepCopy()
-
 	if !r.dynakube.FeatureDisableActivegateRawImage() {
 		err := r.reconcileActiveGateConnectionInfo(ctx)
 		if err != nil {
 			return err
 		}
 	}
+	needStatusUpdate, err := hasher.IsDifferent(oldStatus, r.dynakube.Status)
+	if err != nil {
+		return errors.WithMessage(err, "failed to compare connection info status hashes")
+	} else if needStatusUpdate {
+		err = r.updateDynakubeStatus(ctx)
+	}
+	return err
+}
 
+func (r *Reconciler) ReconcileOA(ctx context.Context) error {
```

Comment:
```suggestion
func (r *Reconciler) ReconcileOoneAgent(ctx context.Context) error {
```
🙏 🙏 🙏

Reply:
just updated the comment to remove a small typo

Reply:
I would prefer several small steps instead of a big-bang change, but only if it is feasible and we do not get side-effects from incrementally changing the code...

Reply:
done